### ISSUE TITLE
Trait

### ISSUE THREAD

**Issue by obi1kenobi** [obi1kenobi/cargo-semver-checks#1423]
### Steps to reproduce the bug with the above code

Consider the following code: [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=e2eac9b1184f08d70503824063ed6389)
```rust
mod private {
    pub trait Super {}
}

/// ```
/// use playground::MaybeSealed;
///
/// struct A;
/// impl MaybeSealed for A {}
/// ```
pub trait MaybeSealed: private::Super {}

impl<T: MaybeSealed> private::Super for T {}
```

### Actual Behaviour

I believe `MaybeSealed` will currently be reported as a sealed trait. This is incorrect, and can cause both false positives and false negatives.

### Expected Behaviour

`MaybeSealed` is not sealed. To confirm this, run the tests in the playground and observe that the doctest compiles and passes without issue.

### Generated System Information

N/A

### Build Configuration

N/A

### Additional Context

Ironically, this is the simplified case listed in the blog post about [resolving cyclic trait relationships](https://predr.ag/blog/when-is-trait-can-be-implemented-public-api/#cyclic-sealing-relationships). It was imprecisely reduced from the correctly reduced repro example in #1076.
